spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/context/properties/ConfigurationPropertiesReportEndpoint.java

```diff
@@ -107,17 +111,36 @@ public void setKeysToSanitize(String... keysToSanitize) {
 		this.sanitizer.setKeysToSanitize(keysToSanitize);
 	}
 
-	@ReadOperation
+	/**
+	 * Retrieves info for all registered configuration properties.
+	 * @return info for all registered configuration properties
+	 * @deprecated in favor of {@link #configurationProperties(String)}
+	 */
 	public ApplicationConfigurationProperties configurationProperties() {
-		return extract(this.context);
+		return configurationProperties(null);
+	}
+
+	@ReadOperation
+	public ApplicationConfigurationProperties configurationProperties(@Nullable String prefix) {
```

Comment:
I chose to keep the filtering somewhat constrained and not implement a generic end-all-be-all regex pattern. This does allow matching a broader prefix as it does a `starts with` on it.